src/renderers/ArtboardRenderer.js

```diff
@@ -18,7 +18,7 @@ export default class ArtboardRenderer extends SketchRenderer {
 
     return {
       _class: 'artboard',
-      do_objectID: generateID(),
+      do_objectID: props.id ? generateID(props.id, true) : generateID(props.name),
```

Comment:
I'm wondering if we need to introduce a new `id` prop 🤔 

I think just have the name and making it the "hardcoded" seed all the time for artboard should be good enough. Maybe we can prefix the name with `artboard:` to make it won't conflict with other layers.

What do you think?

Reply:
I implemented the new `id` prop incase someone has duplicate artboard names, or leaves the default name of `Artboard`. I was trying to avoid cases of duplicate UUIDs, but maybe in that case, there should be a console warning instead? The `id` prop should match the `flow.target` input also, so a layer display name might not necessarily be what someone would want to use in `flow={{ target: 'id-name' }}`.

Maybe I could change it to name though, and leave it as a random UUID if `name` is left as `Artboard`, and leave the `id` prop to override the name if someone wants?

Reply:
Thoughts on this solution? I replace `hardcoded` with an `index` prop to retrieve an ID by index, this means if there's duplicate artboards with the same name, the prototype link will go to the first one.

I think it would be worth using `name`, but with `id` as an optional override if you want a different layer display name. 

`generateID` function:

```js
export function generateID(seed?: ?string, index?: number): string {
  let _seed: ?string;

  if (seed && !index) {
    if (!previousSeeds[seed]) {
      previousSeeds[seed] = 0;
    }
    previousSeeds[seed] += 1;

    _seed = `${seed}${previousSeeds[seed]}`;
  } else if (seed && index) {
    _seed = `${seed}${index}`;
  }

  return e7(_seed);
}
```

`ArtboardRenderer.js`

```js
...
do_objectID: generateID(props.id || props.name, 1),
...
```

Reply:
I was more thinking something like:
```js
export function generateID(seed?: ?string, hardcoded?: boolean): string {
  let _seed: ?string;

  if (seed) {
    if (!previousSeeds[seed]) {
      previousSeeds[seed] = 0;
    }
    previousSeeds[seed] += 1;

    if (!hardcoded) {
      _seed = `${seed}${previousSeeds[seed]}`;
    }
  }

  return e7(_seed);
}
```

and in `ArtboardRenderer.js`

```js
...
do_objectID: generateID(`artboard:${props.name}`, true),
...
```
(so that we are sure it doesn't conflict with other types of layers

Reply:
I prefer having an `id` optional override, so a dev can use `flow={{ target: '/about:mobile' }}` and a designer/Sketch user would see `About | Mobile` in the layers, and using `index` instead of `hardcoded` should work nicer for duplicate artboard names/ids I think?

I can revert to your proposed `generateID` and remove the `id` prop if you'd prefer though. Good point on the `artboard:` prefix 👍

Updated code diff: 
https://github.com/airbnb/react-sketchapp/pull/436/commits/6d0d88708698008991566ad1d01044a253d3c228

Reply:
> so a dev can use flow={{ target: '/about:mobile' }} and a designer/Sketch user would see About | Mobile in the layers

but why would the dev not do `flow={{ target: 'About | Mobile' }}`? It's really Sketch specific anyway so I don't think it matters if it looks like a URL or not, right?

> using index instead of hardcoded should work nicer for duplicate artboard names/ids

But using an `index` hardcoded as `1` instead of just a boolean won't change anything and might actually conflict with an existing `previousSeeds` 🤔

Reply:
I think I'd rather not add a new prop for now so that we don't expand the API surface (a lot harder to remove stuff than adding it), but let's keep it in mind if it turns out to be a pain

Reply:
👍Applied your suggestions. I agree that accepting an `id` prop for `<Artboard>` is messy and could lead people to expect the same prop to work for other components (inconsistency).

Would you have any thoughts on exposing internal API access for things like the seed override to library developers (on the lines of dangerouslySetInnerHTML)?

For `react-sketchapp-router`, I'd like to be able to do `<Link to="/about">` for an artboard with name `"About | Mobile"`, but that should be resolvable with an abstraction of an internal map of paths and artboard names in the `<Link>` and `<Route>` components.

Reply:
The thinking behind using `index` instead of `hardcoded` for `generateID`, was to handle the case of multiple artboards with the same name. As far as I can tell, having `hardcoded` as true for all artboards, means duplicate names result in duplicate object IDs.

With `generateID(target, 1),`, the target falls back to the first Artboard component in the tree.